File: HackerEarth/maxmin.py
n = int(input())
for num in range(n):
    num = 0
    a = [ [-1 for x in range(n)] for y in range(n)]
    while num < n:
        i = 0
        while i < (n+1)//2:
            a[(i+num)%n][num] = num
            i += 1
        j = 0
        while j < (n + 1)//2 + 1:
            a[num][(j+num)%n] = num
            j += 1
        num += 1
    print(a)

    for i in range(n):
        for j in range(j):
            if j == 0:
                continue
        print(a[i][j]+1,end=" ")
        print()

File: HackerEarth/maxmin.cpp
/**
 *    author:  tourist
 *    created: 14.07.2018 20:05:56
**/
#include <bits/stdc++.h>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int n;
  cin >> n;
  vector<vector<int>> a(n, vector<int>(n, -1));
  for (int num = 0; num < n; num++) {
    for (int i = 0; i < (n + 1) / 2; i++) {
      a[(i + num) % n][num] = num;
    }
    for (int j = 0; j < n / 2 + 1; j++) {
      a[num][(j + num) % n] = num;
    }
  }
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        cout<<a[i][j]<<" ";
    cout<<"\n";
    }
  }
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      if (j > 0) {
        cout << ' ';
      }
      cout << a[i][j] + 1;
    }
    cout << '\n';
  }
  return 0;
}